tools/unit-test/slang-unit-test.h

```diff
@@ -3,36 +3,73 @@
 #include "slang.h"
 #include "source/core/slang-render-api-util.h"
 
-namespace slang
+enum class TestResult
 {
-    struct UnitTestContext
-    {
-        slang::IGlobalSession* slangGlobalSession;
-        const char* workDirectory;
-        ISlangWriter* outputWriter;
-        Slang::RenderApiFlags enabledApis;
-    };
-
-    typedef SlangResult (*UnitTestFunc)(UnitTestContext*);
-
-    class IUnitTestModule
-    {
-    public:
-        virtual SlangInt getTestCount() = 0;
-        virtual const char* getTestName(SlangInt index) = 0;
-        virtual UnitTestFunc getTestFunc(SlangInt index) = 0;
-    };
-
-    class UnitTestRegisterHelper
-    {
-    public:
-        UnitTestRegisterHelper(const char* name, UnitTestFunc testFunc);
-    };
-
-    typedef slang::IUnitTestModule* (*UnitTestGetModuleFunc)();
+    // NOTE! Must keep in order such that combine is meaningful. That is larger values are higher precident - and a series of tests that has lots of passes
+    // and a fail, is still a fail overall. 
+    Ignored,
+    Pass,
+    Fail,
+};
+
+enum class TestMessageType
+{
+    Info,                   ///< General info (may not be shown depending on verbosity setting)
+    TestFailure,           ///< Describes how a test failure took place
+    RunError,              ///< Describes an error that caused a test not to actually correctly run
+};
+
+class ITestReporter
```

Comment:
I guess this interface isn't strictly 'COM', but to avoid ABI issues specifying SLANG_NO_THROW and SLANG_MCALL would be a good idea on the methods.